job-server/src/main/scala/spark/jobserver/HealthCheck.scala

```diff
@@ -0,0 +1,81 @@
+package spark.jobserver
+
+import akka.actor.ActorRef
+import akka.pattern.ask
+import akka.util.Timeout
+import org.slf4j.LoggerFactory
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.control.NonFatal
+import spray.httpx.SprayJsonSupport.sprayJsonMarshaller
+import spray.json.DefaultJsonProtocol._
+import spray.http.StatusCodes
+import spray.routing.{HttpService, RequestContext}
+import spark.jobserver.io.JobInfo
+import spark.jobserver.io.JobDAOActor
+
+trait HealthCheckService {
+
+  def healthCheckImpl(ctx: RequestContext, supervisor: ActorRef, jobInfo: ActorRef,
+                      daoActor: ActorRef)
+
+}
+
+class ActorsHealthCheck extends HealthCheckService {
```

Comment:
Since you want to verify that all user created actors are alive, have you considered sending an `Identify` message to all actors e.g. `/users/* ! Identify` ? If you receive response from supervisor, dao, info and result actor then everything is fine. By doing this your code will be simplified and you won't need to pass all the actors inside.

Reply:
@bsikander I tried this approach. But seems like this will work only if the sender is also an actor. In this case, WebApi or HealthCheck is not an actor. So I might have to send message to individual actors.

Reply:
ping @bsikander